design/build-clusters.md

```diff
@@ -0,0 +1,165 @@
+# Multiple Build Cluster 
+
+## Context
+
+Build a scalable Screwdriver build infrastructure. 
+
+## Objective 
+
+1. Running builds in multiple build clusters to support high availability
+2. Users can bring in their own build clusters for their own specific infrastructure needs.
+3. Support build cluster in AWS
+
+## Proposal
+
+1. Implicit build clusters
+	Screwdriver maintains its own multiple build cluster infrastructure.
+
+2. Explicit build clusters
+	Users can bring in their own build cluster infrastructure. 
+	
+This can also be Screwdriver maintained specialized cluster where all users have access. Eg: Mobile iOS cluster or Windows cluster.
+	
+
+As part of cluster onboarding process for above both options cluster admin should register their build cluster information with Screwdriver. Below details are required to register a build cluster.
+
+	1. Cluster name - Cluster name provided by the client
+	2. isActive - Cluster status whether its active or inactive. This will be used to route / pause. Initially this will be a manual update of cluster health.
+	3. Authorization - Authorize using user credentials and queue details.
+	4. SCM Context - git url (github.com). Applicable to only explicit build clusters which are non SD maintained.
+	5. SCM Organization - git organizations. Will be used to validate if job has permission to run on build cluster which is requesting. Applicable to only explicit build clusters which are non SD maintained.
+	6. managedByScrewdriver (true / false) - if true Cluster is maintained by screwdriver team, false cluster is maintained by external team.  
```

Comment:
lowercase `Cluster` -> `cluster`. 

Can change `screwdriver` -> `SD` for consistency.